In [1]:
import pandas as pd
import numpy as np
# import pydot

import matplotlib.pyplot as plt
import matplotlib
import seaborn as sns
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.svm import SVC

from sklearn.metrics import f1_score,roc_curve,auc,precision_score,recall_score

from sklearn.metrics import confusion_matrix

import seaborn as sns
from scipy import stats
import scipy.stats
import statsmodels.api as smd
from statsmodels.stats.proportion import proportions_ztest

# 가설검정, 분산분석하는 패키지
from statsmodels.formula.api import ols # 분산분석하기? 모델링?
from statsmodels.stats.anova import anova_lm # 아노바 데이터프레임 형태 출력
from scipy.stats import shapiro
%matplotlib inline
from sklearn.preprocessing import scale,minmax_scale,robust_scale
from scipy.stats import probplot
from sklearn.model_selection import train_test_split
from statsmodels.api import Logit

# font_path = 'C:/Windows/Fonts/gulim.ttc'
font_path = 'C:/Windows/Fonts/gulim.ttc'
font_name = matplotlib.font_manager.FontProperties(fname=font_path).get_name()
matplotlib.rc('font', family = font_name)

from matplotlib import font_manager,rc

from statsmodels.stats.outliers_influence import variance_inflation_factor
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.preprocessing import StandardScaler
from statsmodels.api import qqplot, add_constant

# from sklearn.tree import export_graphviz
# import graphviz

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error
import numpy as np
# import pydot

import matplotlib.pyplot as plt
import matplotlib
import seaborn as sns
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.svm import SVC

from sklearn.metrics import f1_score,roc_curve,auc,precision_score,recall_score

from sklearn.metrics import confusion_matrix

import seaborn as sns
from scipy import stats
import scipy.stats
import statsmodels.api as smd
from statsmodels.stats.proportion import proportions_ztest

# 가설검정, 분산분석하는 패키지
from statsmodels.formula.api import ols # 분산분석하기? 모델링?
from statsmodels.stats.anova import anova_lm # 아노바 데이터프레임 형태 출력
from scipy.stats import shapiro
%matplotlib inline
from sklearn.preprocessing import scale,minmax_scale,robust_scale
from scipy.stats import probplot
from sklearn.model_selection import train_test_split
from statsmodels.api import Logit

# # font_path = 'C:/Windows/Fonts/gulim.ttc'
# font_path = '/usr/share/fonts/opentype/noto/NotoSerifCJK-ExtraLight.ttc'
# font_name = matplotlib.font_manager.FontProperties(fname=font_path).get_name()
# matplotlib.rc('font', family = font_name)

from matplotlib import font_manager,rc

from statsmodels.stats.outliers_influence import variance_inflation_factor
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.preprocessing import StandardScaler
from statsmodels.api import qqplot, add_constant

# from sklearn.tree import export_graphviz
# import graphviz

from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error

In [2]:
g_high=pd.read_csv('high_danger_g.csv',encoding='euckr').drop('Unnamed: 0',axis=1)
g_low=pd.read_csv('low_danger_g.csv',encoding='euckr').drop('Unnamed: 0',axis=1)
g_profit=pd.read_csv('profit_reject_g.csv',encoding='euckr').drop('Unnamed: 0',axis=1)

In [3]:
b_high=pd.read_csv('high_danger.csv',encoding='euckr').drop('Unnamed: 0',axis=1)
b_low=pd.read_csv('low_danger.csv',encoding='euckr').drop('Unnamed: 0',axis=1)
b_profit=pd.read_csv('profit_reject.csv',encoding='euckr').drop('Unnamed: 0',axis=1)

In [4]:
high = pd.concat([g_high,b_high])
low = pd.concat([g_low,b_low])
profit = pd.concat([g_profit,b_profit])

In [5]:
low['risk'] = 1

In [6]:
high['risk'] = 2

In [7]:
df = pd.concat([high,low,profit])
df.groupby('risk').count()

customer_id
risk             
0.0            83
1.0          2673
2.0            71

In [8]:
df

customer_id  risk
0      C115355   2.0
1      C116232   2.0
2      C120783   2.0
3      C126116   2.0
4      C127039   2.0
..         ...   ...
71     C126909   0.0
72     C127467   0.0
73     C127560   0.0
74     C133934   0.0
75     C134198   0.0

[2827 rows x 2 columns]

In [9]:
df_raw = pd.read_csv('insu_pre_review.csv',encoding = 'euckr')
df_pro = pd.merge(df_raw,df,on='customer_id',how='left')
df_pro = df_pro[df_pro['risk'].isna() == False]
df_pro

customer_id review_cat judge review_date  gender  age  height  weight  \
4         C112358       혈액검사    거절  2015-02-24       1   70     166      73   
12        C112374       혈액검사    거절  2015-01-22       1   70     175      66   
37        C112409       일반검진    거절  2015-01-20       1   69     178      75   
40        C112414       일반검진    거절  2014-10-13       1   68     165      70   
44        C112421       일반검진    거절  2015-08-02       1   69     165      55   
...           ...        ...   ...         ...     ...  ...     ...     ...   
14908     C134201       혈액검사    거절  2015-02-16       1   48     160      58   
14909     C134203       혈액검사    거절  2015-09-07       1   51     165      65   
14910     C134204       혈액검사    거절  2015-03-10       1   54     180      83   
14929     C134237       혈액검사    거절  2014-10-30       2   53     158      62   
14931     C134240       일반검진    거절  2015-08-27       2   22     160      47   

       bust  waist  ...  bt_mchc_judge  bt_mvc_judge  bt_plat_judge  \
4       100     91  ...            0.0           0.0            0.0   
12       98     84  ...            0.0           0.0            0.0   
37       99     90  ...            NaN           NaN            NaN   
40       94     88  ...            NaN           NaN            NaN   
44       93     82  ...            NaN           NaN            NaN   
...     ...    ...  ...            ...           ...            ...   
14908    91     74  ...            0.0           0.0            0.0   
14909    98     87  ...            0.0           0.0            0.0   
14910    95     86  ...            0.0           0.0            0.0   
14929    90     80  ...            0.0           0.0            0.0   
14931    80     60  ...            NaN           NaN            NaN   

       bt_rbc_judge  bt_wbc_judge  bt_rgpt_judge  bt_sgot_judge  \
4               0.0           0.0            0.0            0.0   
12              0.0           0.0            0.0            0.0   
37              NaN           NaN            NaN            NaN   
40              NaN           NaN            NaN            NaN   
44              NaN           NaN            NaN            NaN   
...             ...           ...            ...            ...   
14908           0.0           0.0            0.0            0.0   
14909           0.0           0.0            1.0            0.0   
14910           0.0           0.0            0.0            0.0   
14929           0.0           0.0            0.0            0.0   
14931           NaN           NaN            NaN            NaN   

       bt_sgpt_judge  bt_trig_judge risk  
4                1.0            0.0  1.0  
12               0.0            0.0  1.0  
37               NaN            NaN  1.0  
40               NaN            NaN  1.0  
44               NaN            NaN  1.0  
...              ...            ...  ...  
14908            0.0            0.0  1.0  
14909            0.0            1.0  1.0  
14910            0.0            0.0  1.0  
14929            0.0            0.0  1.0  
14931            NaN            NaN  1.0  

[2827 rows x 51 columns]

In [11]:
df_pro[df_pro['review_cat'] == '혈액검사'].groupby('risk')[['customer_id']].count()

customer_id
risk             
0.0            76
1.0          2204
2.0            64

## 혈액검사 기준 거절 고객(청구 목표변수 설정 시)

- 1분류 결과 : 수익 76명 + 위험 2268명
- 2분류 결과 : 저위험 2204명 + 고위험 64명

In [12]:
df_pro[df_pro['review_cat'] != '혈액검사'].groupby('risk')[['customer_id']].count()

customer_id
risk             
0.0             7
1.0           469
2.0             7

## 일반검진 기준 거절 고객(청구 목표변수 설정 시)

- 1분류 결과 : 수익 7명 + 위험 476명
- 2분류 결과 : 저위험 469명 + 고위험 7명

In [14]:
df_test= df_pro[df_pro['review_cat'] == '혈액검사']
df_test.groupby('risk').count()

customer_id  review_cat  judge  review_date  gender   age  height  \
risk                                                                      
0.0            76          76     76           76      76    76      76   
1.0          2204        2204   2204         2204    2204  2204    2204   
2.0            64          64     64           64      64    64      64   

      weight  bust  waist  ...  bt_mch_judge  bt_mchc_judge  bt_mvc_judge  \
risk                       ...                                              
0.0       76    76     76  ...            76             76            76   
1.0     2204  2204   2204  ...          2204           2204          2204   
2.0       64    64     64  ...            64             64            64   

      bt_plat_judge  bt_rbc_judge  bt_wbc_judge  bt_rgpt_judge  bt_sgot_judge  \
risk                                                                            
0.0              76            76            76             76             76   
1.0            2204          2204          2204           2204           2204   
2.0              64            64            64             64             64   

      bt_sgpt_judge  bt_trig_judge  
risk                                
0.0              76             76  
1.0            2204           2204  
2.0              64             64  

[3 rows x 50 columns]

In [11]:
df_pro.to_csv('req_predict.csv',encoding='euckr')